src/uu/dd/src/parseargs/unit_tests.rs

```diff
@@ -0,0 +1,401 @@
+        Flag::Append,
+        Flag::SeekBytes,
+    ];
+
+    let args = vec![
+        String::from("dd"),
+        String::from("--iflag=fullblock,count_bytes,skip_bytes,direct,directory,dsync,sync,nonblock,noatime,noctty,nofollow,append,seek_bytes"),
+        // String::from("--iflag=fullblock,count_bytes,skip_bytes,cio,direct,directory,dsync,sync,nocache,nonblock,noatime,noctty,nofollow,nolinks,binary,text,append,seek_bytes"),
+    ];
+    let matches = uu_app().get_matches_from_safe(args).unwrap();
+
+    let act = parse_flag_list::<Flag>("iflag", &matches).unwrap();
+
+    assert_eq!(exp.len(), act.len());
+    for cf in &exp {
+        assert!(exp.contains(&cf));
+    }
+}
+
+#[test]
+fn parse_oflag_tokens() {
+    let exp = vec![
+        Flag::FullBlock,
+        Flag::CountBytes,
+        Flag::SkipBytes,
+        // Flag::Cio,
+        Flag::Direct,
+        Flag::Directory,
+        Flag::Dsync,
+        Flag::Sync,
+        // Flag::NoCache,
+        Flag::NonBlock,
+        Flag::NoATime,
+        Flag::NoCtty,
+        Flag::NoFollow,
+        // Flag::NoLinks,
+        // Flag::Binary,
+        // Flag::Text,
+        Flag::Append,
+        Flag::SeekBytes,
+    ];
+
+    let args = vec![
+        String::from("dd"),
+        String::from("--oflag=fullblock,count_bytes,skip_bytes,direct,directory,dsync,sync,nonblock,noatime,noctty,nofollow,append,seek_bytes"),
+        // String::from("--oflag=fullblock,count_bytes,skip_bytes,cio,direct,directory,dsync,sync,nocache,nonblock,noatime,noctty,nofollow,nolinks,binary,text,append,seek_bytes"),
+    ];
+    let matches = uu_app().get_matches_from_safe(args).unwrap();
+
+    let act = parse_flag_list::<Flag>("oflag", &matches).unwrap();
+
+    assert_eq!(exp.len(), act.len());
+    for cf in &exp {
+        assert!(exp.contains(&cf));
+    }
+}
+
+// ----- Multiplier Strings etc. -----
+macro_rules! test_byte_parser (
+    ( $test_name:ident, $bs_str:expr, $bs:expr ) =>
+    {
+        #[allow(non_snake_case)]
+        #[test]
+        fn $test_name()
+        {
+            // let bs_str = String::from($bs_str);
+            assert_eq!($bs, parse_bytes_with_opt_multiplier($bs_str).unwrap())
+        }
+    }
+);
+
+test_byte_parser!(test_bytes_n, "765", 765);
+test_byte_parser!(test_bytes_c, "13c", 13);
+
+test_byte_parser!(test_bytes_w, "1w", 2);
+
+test_byte_parser!(test_bytes_b, "1b", 512);
+
+test_byte_parser!(test_bytes_k, "1kB", 1000);
+test_byte_parser!(test_bytes_K, "1K", 1024);
+test_byte_parser!(test_bytes_Ki, "1KiB", 1024);
+
+test_byte_parser!(test_bytes_MB, "2MB", 2 * 1000 * 1000);
+test_byte_parser!(test_bytes_M, "2M", 2 * 1024 * 1024);
+test_byte_parser!(test_bytes_Mi, "2MiB", 2 * 1024 * 1024);
+
+test_byte_parser!(test_bytes_GB, "3GB", 3 * 1000 * 1000 * 1000);
+test_byte_parser!(test_bytes_G, "3G", 3 * 1024 * 1024 * 1024);
+test_byte_parser!(test_bytes_Gi, "3GiB", 3 * 1024 * 1024 * 1024);
+
+test_byte_parser!(test_bytes_TB, "4TB", 4 * 1000 * 1000 * 1000 * 1000);
```

Comment:
The errors about the overflow is because on 32 bit systems this will overflow an usize. To fix you can add `    #[cfg(target_pointer_width = "64")]` on the offending lines, which will only enable the test on 64 bit.

Reply:
Agreed, however testsdiepraam mentioned that `uucore::parse_size` is an option. I've switched to that instead.

Reply:
Actually, I mis-read this. I have now made this change.